redis/src/cluster_client.rs

```diff
@@ -44,6 +44,12 @@ impl ClusterClientBuilder {
     /// usernames, an error is returned.
     pub fn build(self) -> RedisResult<ClusterClient> {
         let mut initial_nodes = self.initial_nodes?;
+        if initial_nodes.is_empty() {
```

Comment:
Instead of doing it this way, I'd prefer to anchor this in the type system, something like this:

```rust
let first_node = match initial_nodes.first() {
    Some(node) => node,
    None => return Err(..),
};
```

then use `first_node` instead of `initial_nodes[0]` below.

Reply:
You apparently applied these changes in some other place in an earlier commit, but not here.